node/rest/src/routes.rs

```diff
@@ -16,373 +16,212 @@
+        Json(rest.ledger.latest_state_root())
     }
 
-    /// Returns the latest state root.
-    async fn latest_state_root(ledger: Ledger<N, C>) -> Result<impl Reply, Rejection> {
-        Ok(reply::json(&ledger.latest_state_root()))
-    }
+    // GET /testnet3/block/{height}
+    // GET /testnet3/block/{blockHash}
+    pub(crate) async fn get_block(
+        State(rest): State<Self>,
+        Path(height_or_hash): Path<String>,
+    ) -> Result<Json<Block<N>>, RestError> {
+        // Manually parse the height or the height or the hash, axum doesn't support different types
+        // for the same path param.
+        let block = if let Ok(height) = height_or_hash.parse::<u32>() {
+            rest.ledger.get_block(height)?
+        } else {
+            let hash = height_or_hash
+                .parse::<N::BlockHash>()
+                .map_err(|_| RestError("invalid input, it is neither a block height nor a block hash".to_string()))?;
 
-    /// Returns the block for the given block height.
-    async fn get_block(height: u32, ledger: Ledger<N, C>) -> Result<impl Reply, Rejection> {
-        Ok(reply::json(&ledger.get_block(height).or_reject()?))
-    }
+            rest.ledger.get_block_by_hash(&hash)?
+        };
 
-    /// Returns the blocks for the given block range.
-    async fn get_blocks(block_range: BlockRange, ledger: Ledger<N, C>) -> Result<impl Reply, Rejection> {
-        let start_height = block_range.start;
-        let end_height = block_range.end;
+        Ok(Json(block))
+    }
 
+    // GET /testnet3/blocks?start={start_height}&end={end_height}
+    pub(crate) async fn get_blocks(
+        State(rest): State<Self>,
+        Query((start_height, end_height)): Query<(u32, u32)>,
+    ) -> Result<Json<Vec<Block<N>>>, RestError> {
         const MAX_BLOCK_RANGE: u32 = 50;
 
         // Ensure the end height is greater than the start height.
         if start_height > end_height {
-            return Err(reject::custom(RestError::Request("Invalid block range".to_string())));
+            return Err(RestError("Invalid block range".to_string()));
         }
+
         // Ensure the block range is bounded.
-        else if end_height - start_height > MAX_BLOCK_RANGE {
-            return Err(reject::custom(RestError::Request(format!(
+        if end_height - start_height > MAX_BLOCK_RANGE {
+            return Err(RestError(format!(
                 "Cannot request more than {MAX_BLOCK_RANGE} blocks per call (requested {})",
                 end_height - start_height
-            ))));
+            )));
         }
 
         let blocks = cfg_into_iter!((start_height..end_height))
-            .map(|height| ledger.get_block(height).or_reject())
+            .map(|height| rest.ledger.get_block(height))
             .collect::<Result<Vec<_>, _>>()?;
 
-        Ok(reply::json(&blocks))
+        Ok(Json(blocks))
     }
 
-    /// Returns the block for the given block hash.
```

Comment:
All the method doc comments are removed. Intentional?

Reply:
Unintentional, will add back in—good catch!

Reply:
On second thought, I might not add these back in; those functions are now condensed with the route handlers and their functionality is quite clear from the code. I don't think the old docs would add anything. If there's a particular endpoint that is lacking, however, feel free to point that out 🙏